src/particles.cpp

```diff
@@ -142,6 +147,18 @@ void Particle::step(float dtime)
 		m_velocity += m_acceleration * dtime;
 		m_pos += m_velocity * dtime;
 	}
+	if (m_animation.type != TAT_NONE) {
+		m_animation_time += dtime;
+		int frame_length_i, frame_count;
+		m_animation.determineParams(
+				m_material.getTexture(0)->getSize(),
+				&frame_count, &frame_length_i, NULL);
+		float frame_length = frame_length_i / 1000.0;
+		while (m_animation_time > frame_length) {
```

Comment:
Why is a loop here needed? It can be written as
```C++
float count = floor(m_animation_time/frame_length);
m_animation_time -= count * frame_length;
m_animation_frame += count;
```
no?